### PR TITLE
Add IPv6 streaming support

### LINKED ISSUE #234: IPv6 not supported
### Describe the Bug

Currently, IPv6 is not used by any of the network sockets.

If you run `netstat -lnp | grep sunshine`, you can see all of the sockets bound to `0.0.0.0`, not `[::]`. Also, if you look at https://github.com/SunshineStream/Sunshine/blob/master/sunshine/network.cpp, you can see it explicitly only binds with AF_INET.

On the server side, my ISP at puts me behind an IPv4 CGNAT, but has full end-to-end IPv6 connectivity.

And on the client side, both locations where I'd use it (either at work, or via my mobile) are also IPv6–only. (VPNs are not always feasible due to corp policy reasons.)

### Expected Behavior

The Moonlight client supports IPv6. I expected to be able to connect to Sunshine using IPv6, but wasn't able to due to its lack of support.

### Additional Context

_No response_

### Sunshine Host Operating System and Version

Fedora 36

### Architecture

x86_64

### Sunshine Version

0.14.0

### GPU Type

AMD

### GPU Model

RX 6900XT

### GPU Driver/Mesa Version

22.0.1

### Capture Method (Linux Only)

Wayland

### PR DESCRIPTION
**Pull request by cgutman**
## Description
This PR introduces a new "Address Family" option in the "Advanced" settings page which can be used to select between IPv4-only and IPv4+IPv6 modes for streaming. IPv4-only is the default to match the current behavior.

I've also implemented IPv6 Firewall Control via UPnP (based on the GSv6Fwd code) which is supported by some routers. This will automatically open the IPv6 firewall to allow communication when UPnP is enabled, similar to how we create port mappings via UPnP for IPv4. Some routers that don't support UPnP IPv6 firewall control support PCP for pinholes instead, but I will implement that in a future PR. For now, users can open the IPv6 firewall manually in those cases.

The dual address family mode depends on OS support for the `IPV6_V6ONLY` socket option which allows us to have a single socket that can communicate with IPv4 and IPv6 endpoints. There is currently no IPv6-only option due to the need to plumb support for that `IPV6_V6ONLY` option into `Simple-HTTP-Server`. On OSes like OpenBSD that lack support for `IPV6_V6ONLY` , IPv6+IPv4 mode behaves like IPv6-only.

Note: The web UI is dual-stack, however it doesn't accept on-link interface addresses as LAN. Any GUA will be treated as a WAN address even if it's on-link.

### Screenshot
Screenshot from 2022-12-16 18-27-26

### Issues Fixed or Closed

- Fixes #234 
- Fixes #390 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/network.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace net {
 std::pair<std::uint32_t, std::uint32_t> ip_block(const std::string_view &ip);
 
 std::vector<std::pair<std::uint32_t, std::uint32_t>> pc_ips {
-  ip_block("127.0.0.1/32"sv)
+  ip_block("127.0.0.0/8"sv)
```
Comment (cgutman) [inline src/network.cpp:13]: This was a little drive-by fix. IPv4 loopback is the whole `127.0.0.0/8` subnet, not just `127.0.0.1`.

**Comment by psyke83** [comment]
Tested on my IPv6 router:

* On server: enabled IPv4+IPv6
* On client: log of `flatpak run com.moonlight_stream.Moonlight` shows both IPv4 and IPv6 addresses; connects fine.
* On client with IPv4 temporarily disabled (`ip address remove ...`): log shows IPv6 address only; connects fine.

**Comment by psyke83** [comment]
Update: there seems to be a problem with establishing a connection after disconnecting and reconnecting multiple times (testing with mixed IPv4+IPv6 on client).

The Moonlight log doesn't mention any errors related to connecting, but the GUI pops up this message:

`Starting control stream establishment failed: Error 110
Check your firewall and port forwarding rules for port(s) UDP 47999`

It seems that I can't connect the server again until I manually restart the service.

Server is Windows 10 22H2, client is Ubuntu 22.10.

**Comment by i8372370** [comment]
好耶

**Comment by BigBrainAFK** [comment]
The WebUI can be made accessible via IPv6 if we refer to ULA specified in RFC 4913 which gives a range of non globally routable IPv6 addresses and how they are structured.

**Comment by dongzheng9** [comment]
Web UI: [127.0.0.1] -- not authorized        TAT

**Comment by euroclydon37** [comment]
I know this is a PR and not an issue or discussion, but I just wanted to say thank you for working on this and I wish I had the adequate CPP knowledge to contribute.

File: src/network.cpp (hunk 1 above)
Comment (euroclydon37) [reply src/network.cpp:13]: I love that we have ~16 million loopback addresses. lol

**Comment by euroclydon37** [comment]
Also, I built a beefy server with 2 gpus and I'm passing them each through to VMs and letting a couple friends use them remotely Stadia-style. Without IPv6 support, I'll have to do some much-more-complicated networking configuration.

All that to say that I'm totally willing to do some testing to help this along when you're ready.

Again, thank you a ton for working on this and I'm sure you're prioritizing a lot of different work streams. So no pressure whatsoever. Just expressing interest in the feature and trying to be of some assistance. lol

**Comment by cgutman** [comment]
Now that the major dependency of IPv6 is done in https://github.com/LizardByte/Sunshine/pull/1569 and new versions of the [GameStream IPv6 Forwarder](https://github.com/moonlight-stream/GS-IPv6-Forwarder/releases) and [Moonlight Internet Hosting Tool](https://github.com/moonlight-stream/Internet-Hosting-Tool/releases) are shipped that no longer interfere with IPv6 on the default GameStream ports, IPv6 support should be ready now.

**Comment by euroclydon37** [comment]
You're the best!
